### PR TITLE
Avijit/build script update

### PR DESCRIPTION
**Pull request by avijit-nervana**

### CODE REVIEW

File: build_ngtf.py [hunk 1]
```diff
@@ -18,6 +18,34 @@
...
+def version_check(use_prebuilt_tensorflow):
+    # Check pre-reqisites
+    if use_prebuilt_tensorflow:
+        # Check if the gcc version is 4.8
+        if (platform.system() != 'Darwin'):
+            gcc_ver = get_gcc_version()
+            if '4.8' not in gcc_ver:
+                raise Exception(
+                    "Need GCC 4.8 to build using prebuilt TensorFlow\n"
+                    "Gcc version installed: " + gcc_ver)
```
Comment (sayantan-nervana) [inline build_ngtf.py]: We can add more lines here that say. "Please change gcc to 4.8 OR build Tensorflow from source by skipping the use_prebuilt_tensorflow option"

File: build_ngtf.py [hunk 2]
```diff
@@ -18,6 +18,34 @@
 from tools.build_utils import *
 
 
+def version_check(use_prebuilt_tensorflow):
+    # Check pre-reqisites
```
Comment (sayantan-nervana) [inline build_ngtf.py]:
```suggestion
    # Check pre-requisites
```

File: build_ngtf.py [hunk 3]
```diff
@@ -227,36 +260,75 @@ def main():
...
+                             target_arch, verbosity)
+
+            # Now build the libtensorflow_cc.so - the C++ library
+            build_tensorflow_cc(tf_src_dir, artifacts_location, target_arch,
+                                verbosity)
+
+            # Install tensorflow to our own virtual env
+            # Note that if gcc 4.8 is used for building TensorFlow this flag
+            # will be 0
+            cxx_abi = install_tensorflow(venv_dir, artifacts_location)
```
Comment (sayantan-nervana) [inline build_ngtf.py]: we can add an assert the comment that, `use_prebuilt_tensorflow implies cxx_abi==0`:
`assert (!arguments.use_prebuilt_tensorflow) or (arguments.use_prebuilt_tensorflow and cxx_abi==0), "Expected cxx_abi to be 0 when use_prebuilt_tensorflow is on"`

File: build_ngtf.py [hunk 4]
```diff
@@ -278,6 +350,7 @@ def main():
         "-DNGRAPH_USE_CXX_ABI=" + cxx_abi,
         "-DNGRAPH_DEX_ONLY=TRUE",
         "-DNGRAPH_DEBUG_ENABLE=NO",
+        "-DNGRAPH_UNIT_TEST_ENABLE=NO",
```
Comment (sayantan-nervana) [inline build_ngtf.py]: good one.. maybe it will save some build time

File: build_ngtf.py [hunk 5]
```diff
@@ -344,16 +411,31 @@ def main():
...
+    if arguments.use_tensorflow_from_location:
+        ngraph_tf_cmake_flags.extend([
+            "-DTF_SRC_DIR=" + os.path.abspath(
+                arguments.use_tensorflow_from_location + '/tensorflow')
+        ])
+    else:
+        # Download TF source if it doesn't exist
+        if not os.path.isdir(tf_src_dir):
+            # Download TF source
+            pass
```
Comment (sayantan-nervana) [inline build_ngtf.py]: This `if` is empty

File: README.md [hunk 6]
```diff
@@ -27,44 +27,20 @@ a variety of nGraph-enabled backends: CPU, GPU, and custom silicon like the
...
-### :warning: Not Yet vailable for this version 
+### Option 1: Use a pre-built nGraph-TensorFlow bridge 
 
 1. You can install TensorFlow and nGraph using `virtualenv` or in the system Python location. 
 
-2. Install TensorFlow v1.14.0rc0:
+2. Install TensorFlow v1.14.0:
 
-        pip install -U tensorflow==1.14.0rc0
+        pip install -U tensorflow==1.14.0
```
Comment (sayantan-nervana) [inline README.md]: Have we tested with 1.14.0? All ci is on `tensorflow==v1.14.0-rc0` right now I think

File: build_ngtf.py [hunk 7]
```diff
@@ -227,36 +260,75 @@ def main():
...
+            os.chdir(pwd_now)
+
+            # Next install the tensorflow python packge
             command_executor(
                 ["pip", "install", "-U", "tensorflow==" + tf_version])
             cxx_abi = get_tf_cxxabi()
+
+            # Copy the libtensorflow_framework.so to the artifacts so that
+            # we can run c++ tests from that location later
+            tf_fmwk_lib_name = 'libtensorflow_framework.so.1'
```
Comment (shresthamalik) [inline build_ngtf.py]: Question: I see that we are not building this 'libtensorflow_framework.so.1'. Where are we getting this from?

Comment (avijit-nervana) [reply build_ngtf.py]: This would come from the preinstalled TF.

**Pull request merged**